app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/AgendaViewModel.kt

```diff
@@ -106,6 +103,27 @@ class AgendaViewModel @Inject constructor(
                 )
             }
 
+            // Fetch/Refresh the previous and coming week's agenda items
+            withContext(Dispatchers.IO) {
+                (-5..5).map { index ->
+                    if (index != 0) {
+                        val date = getDateForSelectedDayIndex(selectedDate, index)
+                        async { agendaRepository.getAgendaForDayFromRemote(date) }
+                    } else {
+                        async { null }
+                    }
+                }.awaitAll()
```

Comment:
One thing to note about `awaitAll()` is that it fails as soon as one of the coroutines fails. To prevent that, you'd need another map block:
```
.map { it.await() }
```

Reply:
Interesting. OK, I will refactor and make notes